packages/bibtex/support/bib2csl.lua

```diff
@@ -0,0 +1,178 @@
+   inbook = 'chapter', -- (1)
+   -- bookinbook ?
+   -- suppbook ?
+   booklet = 'pamphlet', -- (1)
+   -- collection ?
+   -- mvcollection ?
+   incollection = 'chapter', -- (1)
+   -- suppcollection ?
+   dataset = 'dataset',
+   manual = 'book', -- (1)
+   misc = 'document',
+   online = 'webpage', -- (1)
+   patent = 'patent', -- (1)
+   periodical = 'article-journal', -- Not sure, (1) has book
+   -- suppperiodical ?
+   proceedings = 'book', -- (1)
+   inproceedings = 'paper-conference', -- (1)
+   reference = 'entry-dictionary',
+   mvreference = 'entry-dictionary',
+   report = 'report', -- (1) via aliases
+   -- set [special case]
+   software = 'software',
+   thesis = 'thesis', -- (1) via aliases
+   unpublished = 'manuscript', -- (1)
+
+   -- BibLaTeX manual v3.20, §2.1.2 (aliases)
+   conference = 'event', -- not sure, should be equivalent to @inproceedings
+   electronic = 'webpage', -- as @online
+   mastersthesis = 'thesis', -- as @thesis
+   phdthesis = 'thesis', -- as @thesis
+   techreport = 'report', -- as @report
+   www = 'webpage', -- as @online
+
+   -- BibLaTeX manual v3.20, §2.1.3 (non-standard)
+   -- artwork ?
+   -- audio ?
+   -- bibnote [special case]
+   -- commentary ?
+   -- image ?
+   -- jurisdiction ?
+   legislation = 'legislation',
+   -- legal ?
+   -- letter ?
+   -- movie ?
+   -- music ?
+   -- performance ?
+   -- review ?
+   standard = 'legislation', -- (1)
+   -- video ?
+}
+
+local function toDate (year, month)
+   local date = {year = year, month = month}
+   return date
+end
+
+--- Convert a BibTeX entry to a CSL item.
+-- @tparam table entry The BibTeX entry
+-- @treturn table The CSL item
+local function bib2csl (entry)
+   local csl = {}
+
+   -- BibTeX type
+   local t = BIBTEX2CSL_TYPES[entry.type] or 'document'
+
+   csl.type = t
+
+   -- BibTeX address / BibLaTeX location
+   if entry.location then
+      csl['event-place'] = entry.location
+      csl['publisher-place'] = entry.location
+   else
+      csl['event-place'] = entry.address
+      csl['publisher-place'] = entry.address
+   end
+
+   -- BibTeX author
+   csl.author = entry.author
+
+   -- BibTex editor
+   csl.editor = entry.editor
+   csl['collection-editor'] = entry.editor
+
+   -- BibLaTeX date / BibTeX year and month
+   local date = entry.date and entry.date or toDate(entry.year, entry.month)
+   csl.issued = date
+
+   -- BibLaTeX eventdate [BibTeX date]
+   csl['event-date'] = entry.eventdate and toDate(entry.eventdate) or date
+
+   -- BibLaTeX urldate
+   csl.accessed = entry.urldate
+
+   -- BibLaTeX origdate
+   csl['original-date'] = entry.origdate
+
+   -- BibTeX volume
+   csl.volume = entry.volume
+   -- BibLaTeX volumes
+   csl['number-of-volumes'] = entry.volumes
+
+   -- BibTeX edition -- FIXME Can be a literal string or a number
+   csl.edition = entry.edition
+   -- BibTeX version
+   csl.version = entry.revision
+
+   -- BibTeX number and issue
+   csl.number = entry.number
+   csl.issue = entry.issue
```

Comment:
Not so trivial maybe:
https://github.com/JabRef/jabref/issues/8372#issuecomment-1023768144

Reply:
Reproduced, indeed. Fixed minimally as suggested in the above link. Seems to do the trick.